Common/Systems/VanillaModifications/AddClassSelectInPlayer.cs

```diff
@@ -0,0 +1,69 @@
+﻿using PathOfTerraria.Common.Classing;
+using PathOfTerraria.Common.Systems.ModPlayers;
+using PathOfTerraria.Content.Items.Gear.Weapons.Bow;
+using PathOfTerraria.Content.Items.Gear.Weapons.Wand;
+using PathOfTerraria.Content.Skills.Magic;
+using PathOfTerraria.Content.Skills.Melee;
+using PathOfTerraria.Content.Skills.Ranged;
+using PathOfTerraria.Content.Skills.Summon;
+using System.Runtime.CompilerServices;
+using Terraria.GameContent.UI.States;
+using Terraria.ID;
+
+namespace PathOfTerraria.Common.Systems.VanillaModifications;
+
+internal class AddClassSelectInPlayer : ModSystem
+{
+	public override void Load()
+	{
+		On_UICharacterCreation.FinishCreatingCharacter += FinishCreatingChar;
+		On_UICharacterCreation.SetupPlayerStatsAndInventoryBasedOnDifficulty += ReplaceWeaponWithClassItem;
+	}
+
+	private void ReplaceWeaponWithClassItem(On_UICharacterCreation.orig_SetupPlayerStatsAndInventoryBasedOnDifficulty orig, UICharacterCreation self)
+	{
+		orig(self);
+
+		Player player = GetPlayer(self);
+		StarterClasses classType = player.GetModPlayer<ClassingPlayer>().Class;
+
+		player.GetModPlayer<SkillCombatPlayer>().TryAddSkill(classType switch
+		{
+			StarterClasses.Melee => new Berserk(),
+			StarterClasses.Ranged => new RainOfArrows(),
+			StarterClasses.Summon => new FlameSage(),
+			StarterClasses.Magic or _ => new Fireball(),
+		}, true);
```

Comment:
This is quite a remote location to define properties of starter classes in. It would be preferable to have them be data-driven, with a single system initializing structs containing starter items, starter skills' singleton objects, etc. Does not sound complex. Else just move this portion of logic into `StarterClasses.cs`.